### DISCUSSION TITLE
New building: Caravanserai

### DISCUSSION THREAD

**Discussion by dvincent56** [Keriew/augustus#288] [Ideas]
Hi,
I am new here, I m a french 'javascript' developer (don't throw stones at me) and I am excited to discover Augustus.
I got the sources of the project to take a look and I had fun adding a new building.

Let me show you, "The Caravanserai"

![caravanserai_Ingame](https://user-images.githubusercontent.com/81165266/112065354-d1560c80-8b64-11eb-8af5-71966c8b1da9.png)

Merchant caravans take advantage of this resting place. Goods are bought 10% more expensive and sold 10% cheaper.

![caravanseraiWidget](https://user-images.githubusercontent.com/81165266/112065426-e5017300-8b64-11eb-873f-387b96151a73.png)

I quickly created a visual with paint ... I m not a designer ;)

The building must be unique like the Senate, costs 2500 gold and 20 gold/month. The desirability is negative and require 20 people as workforce.

I have 2 issues left, I don't know why but I can't manage to have a workforce on it (0/20). And I want to add a full workforce condition to get the bonus of 10% instead of just the building presence in the city.

I await your feedback,

Regards.

**Comment by Keriew** [comment]
The problem with labour probably is that building doesn't have labour category defined in CATEGORY_FOR_BUILDING_TYPE array in src/city/labour.c

**Comment by Areldir** [comment]
Very cool idea! No need to worry about the assets for it, I'll have a look at creating a fitting new custom asset from scratch.

For ease of discussion, do you have Discord? Our group primarily discusses development over there.

**Comment by RuoanJi** [comment]
@dvincent56 I ran into the exact same issue with workforce and managed to solve it.

In addition to @Keriew 's suggestion, you also need to create a function called `spawn_figure_caravanserai` in `src/building/figure.c`, and invoke it in `building_figure_generate()`. It looks like your building does not generate any walker, so you just need the labor seeker.

**Reply by dvincent56** [reply]
Thanks a lot for your answer. I will try it this evening.

**Reply by dvincent56** [reply]
I confirm it's working now

**Reply by RuoanJi** [reply]
Good to know!

**Comment by Areldir** [comment]
@dvincent56 if you are happy with adding the caravanserai as a building in Augustus, open a Pull Request and we'll gladly incorporate it.
